src/wp-includes/class-wp-query.php

```diff
@@ -4714,11 +4714,22 @@ public function generate_postdata( $post ) {
 
 		$authordata = get_userdata( $post->post_author );
 
-		$currentday   = mysql2date( 'd.m.y', $post->post_date, false );
-		$currentmonth = mysql2date( 'm', $post->post_date, false );
-		$numpages     = 1;
-		$multipage    = 0;
-		$page         = $this->get( 'page' );
+		$post_date = $post->post_date;
+		if ( ! empty( $post_date ) && '0000-00-00 00:00:00' !== $post_date ) {
+			$month = substr( $post_date, 5, 2 );
+			$day   = substr( $post_date, 8, 2 );
+			$year  = substr( $post_date, 0, 2 );
```

Comment:
I think this should be `2,2` the year is four-digit in database and we need last two digits, this nabs first two (and makes unit tests fail currently).

Reply:
```suggestion
			$year  = substr( $post_date, 2, 2 );
```